Process_Library/GOCART2G_Process.F90

```diff
@@ -93,10 +98,337 @@ module  GOCART2G_Process
 !                   physics and chemistry code into a single process library that only uses
 !                   intrinsic Fortran functions.
 !
+!  01Apr2021  R.Montuoro/NOAA - Added FENGSHA dust scheme and related methods.
+!
+!
 !EOP
 !-------------------------------------------------------------------------
 CONTAINS
 
+!=====================================================================================
+!BOP
+!
+! !IROUTINE:  DustAerosolDistributionKok - Compute Kok's dust size aerosol distribution
+!
+! !INTERFACE:
+   subroutine DustAerosolDistributionKok ( radius, rLow, rUp, distribution )
+
+! !USES:
+   implicit NONE
+
+! !INPUT PARAMETERS:
+   real, dimension(:), intent(in)  :: radius      ! Dry particle bin effective radius [um]
+   real, dimension(:), intent(in)  :: rLow, rUp   ! Dry particle bin edge radii [um]
+
+! !OUTPUT PARAMETERS:
+   real, dimension(:), intent(out) :: distribution    ! Normalized dust aerosol distribution [1]
+
+! !DESCRIPTION: Computes lognormal aerosol size distribution for dust bins according to
+!               J.F.Kok, PNAS, Jan 2011, 108 (3) 1016-1021; doi:10.1073/pnas.1014798108
+!
+! !REVISION HISTORY:
+!
+! 22Feb2020 B.Baker/NOAA    - Original implementation
+! 01Apr2021 R.Montuoro/NOAA - Refactored for GOCART process library
+!
+
+! !Local Variables
+   integer :: n, nbins
+   real    :: diameter, dlam, dvol
+
+! !CONSTANTS
+   real, parameter    :: mmd    = 3.4          ! median mass diameter [um]
+   real, parameter    :: stddev = 3.0          ! geometric standard deviation [1]
+   real, parameter    :: lambda = 12.0         ! crack propagation length [um]
+   real, parameter    :: factor = 1.e0 / (sqrt(2.e0) * log(stddev))  ! auxiliary constant
+
+   character(len=*), parameter :: myname = 'DustAerosolDistributionKok'
+
+!EOP
+!-------------------------------------------------------------------------
+!  Begin...
+
+   distribution = 0.
+
+!  Assume all arrays are dimensioned consistently
+   nbins = size(radius)
+
+   dvol = 0.
+   do n = 1, nbins
+     diameter = 2 * radius(n)
+     dlam = diameter/lambda
+     distribution(n) = diameter * (1. + erf(factor * log(diameter/mmd))) &
+                     * exp(-dlam * dlam * dlam) * log(rUp(n)/rLow(n))
+     dvol = dvol + distribution(n)
```

Comment:
@adarmenov - Not sure I understand your question. Could you please clarify?

Reply:
@rmontuoro Let me elaborate... lines 156-163 are a fast and likely imprecise way to numerically integrate over the size distribution. I was pointing out that the integration needs to be done once, hence one can try to be numerically more precise, e.g. use more than 5 sub-bins to approximate the integral.

Reply:
@adarmenov - Thank you for the clarification, I see your point. I will bring this up with NOAA/ARL, where the scheme was developed. For the time being, however, I don't expect any changes since we would like to compare the performance of the FENGSHA scheme in GOCART with the operational version, which implements this integration technique.